deps/baseconfig/src/typeoperators.h

```diff
@@ -35,20 +35,21 @@
-__forceinline constexpr ENUMTYPE operator^(ENUMTYPE const a, ENUMTYPE const b) { return ENUMTYPE(((SizedInteger<ENUMTYPE>::type)a) ^ ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline ENUMTYPE &operator^=(ENUMTYPE &a, ENUMTYPE const &b) { return (ENUMTYPE &)(((SizedInteger<ENUMTYPE>::type &)a) ^= ((SizedInteger<ENUMTYPE>::type &)b)); } \
-__forceinline ENUMTYPE &operator&=(ENUMTYPE &a, ENUMTYPE const &b) { return (ENUMTYPE &)(((SizedInteger<ENUMTYPE>::type &)a) &= ((SizedInteger<ENUMTYPE>::type &)b)); } \
-__forceinline ENUMTYPE &operator|=(ENUMTYPE &a, ENUMTYPE const &b) { return (ENUMTYPE &)(((SizedInteger<ENUMTYPE>::type &)a) |= ((SizedInteger<ENUMTYPE>::type &)b)); } \
-__forceinline constexpr ENUMTYPE operator<<(ENUMTYPE a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) << ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline constexpr ENUMTYPE operator>>(ENUMTYPE a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) >> ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline ENUMTYPE &operator<<=(ENUMTYPE &a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE &)((SizedInteger<ENUMTYPE>::type &)a = ((SizedInteger<ENUMTYPE>::type &)a) << ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline ENUMTYPE &operator>>=(ENUMTYPE &a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE &)((SizedInteger<ENUMTYPE>::type &)a = ((SizedInteger<ENUMTYPE>::type &)a) >> ((SizedInteger<ENUMTYPE>::type)b)); } \
+__forceinline constexpr ENUMTYPE operator|(ENUMTYPE const a, ENUMTYPE const b) { return ENUMTYPE(((UNDERLYING)a) | ((UNDERLYING)b)); } \
+__forceinline constexpr ENUMTYPE operator&(ENUMTYPE const a, ENUMTYPE const b) { return ENUMTYPE(((UNDERLYING)a) & ((UNDERLYING)b)); } \
+__forceinline constexpr ENUMTYPE operator~(ENUMTYPE const a) { return ENUMTYPE(~((UNDERLYING)a)); } \
+__forceinline constexpr ENUMTYPE operator^(ENUMTYPE const a, ENUMTYPE const b) { return ENUMTYPE(((UNDERLYING)a) ^ ((UNDERLYING)b)); } \
+__forceinline ENUMTYPE &operator^=(ENUMTYPE &a, ENUMTYPE const &b) { return (ENUMTYPE &)(((UNDERLYING &)a) ^= ((UNDERLYING &)b)); } \
+__forceinline ENUMTYPE &operator&=(ENUMTYPE &a, ENUMTYPE const &b) { return (ENUMTYPE &)(((UNDERLYING &)a) &= ((UNDERLYING &)b)); } \
+__forceinline ENUMTYPE &operator|=(ENUMTYPE &a, ENUMTYPE const &b) { return (ENUMTYPE &)(((UNDERLYING &)a) |= ((UNDERLYING &)b)); } \
+__forceinline constexpr ENUMTYPE operator<<(ENUMTYPE a, size_t const b) { return (ENUMTYPE)(((UnsignedInteger<ENUMTYPE>::type)a) << b); } \
+__forceinline constexpr ENUMTYPE operator>>(ENUMTYPE a, size_t const b) { return (ENUMTYPE)(((UnsignedInteger<ENUMTYPE>::type)a) >> b); } \
+__forceinline ENUMTYPE &operator<<=(ENUMTYPE &a, size_t const b) { return (ENUMTYPE &)((UnsignedInteger<ENUMTYPE>::type &)a <<= b); } \
+__forceinline ENUMTYPE &operator>>=(ENUMTYPE &a, size_t const b) { return (ENUMTYPE &)((UnsignedInteger<ENUMTYPE>::type &)a >>= b); } \
```

Comment:
Now shifting is always treated as unsigned type. We could treat it as `UNDERLYING`, but I don't know if that makes sense. I think `UnsignedInteger<ENUMTYPE>` is safe for now.